plugin/trino-clickhouse/src/main/java/io/trino/plugin/clickhouse/expression/SupportsPushdownPredicate.java

```diff
@@ -0,0 +1,45 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.clickhouse.expression;
+
+import com.clickhouse.data.ClickHouseColumn;
+import com.clickhouse.data.ClickHouseDataType;
+import io.trino.plugin.base.expression.ConnectorExpressionRule.RewriteContext;
+import io.trino.plugin.jdbc.JdbcColumnHandle;
+import io.trino.plugin.jdbc.JdbcTypeHandle;
+import io.trino.spi.TrinoException;
+import io.trino.spi.expression.Variable;
+
+import java.util.function.BiPredicate;
+
+import static io.trino.plugin.jdbc.JdbcErrorCode.JDBC_ERROR;
+
+public class SupportsPushdownPredicate
```

Comment:
nit:  Predicate being pushed down and `BiPredicate` are a bit confusing. Can we rename it as `PushdownHandler` or something similar -

Reply:
Agree that it may be misleading about actual implementation. 
`PushdownHandler` should `handle` something, but not `test`.
Considering all other names which come to my mind and what were proposed by AI don't look better.
Leaving it as it is.

Reply:
Renamed to SupportsPushdown